test/build.spec.ts

```diff
@@ -292,8 +292,9 @@ describe('Invalid build input', () => {
 			.then(() => {
 				throw new Error('Error not thrown on null buildStream input');
 			})
-			.catch(BuildProcessError, () => {
+			.catch((e: any): e is BuildProcessError => {
```

Comment:
hmm I don't think this achieves the same... As coded above, I think it will catch any errors of any type, and will cast the error (of any type) to `BuildProcessError` -- i.e. simply disabling type checking.

I believe that the change that would preserve the Bluebird behavior of only catching `BuildProcessError` would be to replace the two catch blocks with a single one as follows:

``` 
	.catch((e: Error) => {
		if (!(e instanceof BuildProcessError)) {
			throw new Error(
				'Incorrect error thrown on null buildStream input: ' + e,
			);
		}
	});
```

Reply:
> I don't think this achieves the same...

Confirmed by looking at the compiled code (Typescript to Javascript, `npm run buld:test`, looking at `build/test/build.spec.js`). The following is not good: :-)

```
        return build_1.runBuildTask(task, docker, secretMap, buildVars)
            .then(() => {
            throw new Error('Error not thrown on null buildStream input');
        })
            .catch((e) => {
            return;
        })
            .catch(e => {
            throw new Error('Incorrect error thrown on null buildStream input: ' + e);
        });
```

Reply:
:facepalm: